pkg/controller/deployment/deployment_controller.go

```diff
@@ -560,8 +556,48 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 
 	// Sync deployment status
 	return dc.syncDeploymentStatus(allRCs, newRC, deployment)
+}
 
-	// TODO: raise an event, neither scaled up nor down.
+func (dc *DeploymentController) syncPausedDeployment(deployment extensions.Deployment) error {
+	newRC, oldRCs, err := dc.getNewRCAndOldRCs(deployment)
+	if err != nil {
+		return err
+	}
+	if len(oldRCs) == 0 {
+		// No old rcs with pods; scale new rc to the full count.
+		_, err := dc.scaleRCAndRecordEvent(newRC, deployment.Spec.Replicas, deployment)
+		return err
+	}
+	allRCs := append(oldRCs, newRC)
+	// Sort
+	sort.Sort(controller.ByAvailableReplicas(allRCs))
+	maxSurge, isPercent, err := util.GetIntOrPercentValue(&deployment.Spec.Strategy.RollingUpdate.MaxSurge)
```

Comment:
If the strategy of deployment is not RollingUpdate, what will happen here? panic for nil pointer? Or Recreate type deployment will never pause so that `syncPausedDeployment ` will never called? I don't know how deployment.spec.paused works, but I feel nervous here.